File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.cpp
class Solution {
public:
    int maxProfit(vector<int>& p) {
        int n = p.size();
        int l = 0, r = l+1;

        int ans = 0;
        while(r < n){
            if (p[l] < p[r]){
                ans = max(ans, p[r] - p[l]);

            } else{
                l = r;
                
            }
            r++;
        }
        return ans;
    }
};

File: Data Structures & Algorithms/is-anagram/submission-2.java
class Solution {
    private int[] charFreq(String a) {
        int[] ans = new int[26];

        for (int i = 0; i < a.length(); i++) {
            int idx = a.charAt(i) - 'a';
            ans[idx]++;
        }
        
        return ans;
    }
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) {
            return false;
        }

        int[] freqS = charFreq(s);
        int[] freqT = charFreq(t);

        for (int i = 0; i < 26; i++) {
            if (freqS[i] != freqT[i]) {
                return false;
            }
        }

        return true;

    }
}